### PR TITLE
Fetch Google Books data through servlet

### PR DESCRIPTION
**Pull request by kashiish**
These commits create a new servlet called GoogleBooksServlet that fetches books data with the parameters passed from JavaScript fetch requests. I updated the JavaScript fetch calls from the API to the new servlet, so the API key is no longer exposed in the front end. Instead I stored the key in an environment variable in Cloud Shell. 

I also added a Jasmine test for the load more books function. 

I will update mylists.js after my other PR is merged so that it makes a request to the servlet instead of the API directly. I will also work on the servlet test later as it will take me some time to figure out how exactly to test it. We needed to hide the API key ASAP, so I thought it would be better to submit a PR without a test for now.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/GoogleBooksServlet.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        if(author != null) {
+            params += "+inauthor:" + author;
+        }
+
+        if(startIndex >= 0) {
+            params += "&startIndex=" + startIndex;
+        }
+
+        URL url = new URL("https://www.googleapis.com/books/v1/volumes?" + params + "&key=" + System.getenv("GOOGLE_API_KEY"));
```
Comment (betrevisan) [inline src/main/java/com/google/sps/servlets/GoogleBooksServlet.java:65]: This looks good

File: src/main/java/com/google/sps/servlets/GoogleBooksServlet.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```
Comment (betrevisan) [inline src/main/java/com/google/sps/servlets/GoogleBooksServlet.java:66]: What is a connection exactly?

File: src/main/java/com/google/sps/servlets/GoogleBooksServlet.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        String line;
+        StringBuffer output = new StringBuffer();
+
+        while ((line = stream.readLine()) != null) {
+            output.append(line);
+        }
+
+        stream.close();
+
+        JsonObject jsonObject = new JsonParser().parse(output.toString()).getAsJsonObject();
```
Comment (betrevisan) [inline src/main/java/com/google/sps/servlets/GoogleBooksServlet.java:79]: Can't gson be used here?

File: src/main/java/com/google/sps/servlets/GoogleBooksServlet.java (hunk 2 above)
Comment (kashiish) [reply src/main/java/com/google/sps/servlets/GoogleBooksServlet.java:66]: It's kind of like `fetch` in Javascript. It allows us to make a request to a URL.

File: src/main/java/com/google/sps/servlets/GoogleBooksServlet.java (hunk 3 above)
Comment (kashiish) [reply src/main/java/com/google/sps/servlets/GoogleBooksServlet.java:79]: I don't think you can use .toJson() on Strings. JsonParser and JsonObject are part of the Gson library.

**Pull request merged**